packages/types/src/json.ts

```diff
@@ -41,3 +41,62 @@ export interface JsonArray extends Array<JsonValue> {}
  * @public
  */
 export type JsonValue = JsonObject | JsonArray | JsonPrimitive;
+
+/**
+ * Attempts to merge two JsonObjects together. In the case of collisions, this function
+ * prefers values from b, unless the value is an object, in which case it recursively
+ * merges the values.
+ *
+ * @param a - The base object
+ * @param b - The object to merge into a
+ * @returns The merged object
+ * @public
+ */
+export const mergeJson = (a: JsonObject, b: JsonObject): JsonObject => {
```

Comment:
No need for this actually, the `ConfigReader` has a perfectly good merging implementation already 😁